### ISSUE TITLE
crashing after new enhancements module update

### ISSUE THREAD

**Issue by Blunderless** [elishacloud/Silent-Hill-2-Enhancements#738]
my game was working perfect for the last two years, now it's crashing all of over the place after i got a little popup that there is an update. now if i take any in-game action; opening doors, checking inventory, standing idle, then it will very likely crash. 

edit: also certain parts of my game seem to be sped up. like my character movement, scrolling through inventory screens, and in-game cutscenes.

anyone know any fixes? can i undo the last update? (whatever it was made to do besides fuck up my game)

**Comment by Blunderless** [comment]
> Please attached your `d3d8.log` file. Open the Config Tool (`SH2EEconfig.exe`) and start disable features, launching the game in between disabling things, to see if a certain feature is causing the issues for you.

well that's really stupid that these issues were non-existent prior to this update, and now i can't use the features that made this version worthwhile

**Comment by elishacloud** [comment]
@Blunderless, we are not experiencing the same issues you are so we need your help to fix the issues.  As @Polymega mentioned, attaching your log file will help us troubleshoot the issue.  Also, if the issue is caused by a specific feature then knowing which feature caused the issue will help us isolate the issue and fix the issue.

Can you try disabling some of the newer features to see if the issue is in one of these features?  If you figure out which feature has the issue please let us know so we can fix it.

Here are some suggestions for features to try turning off:
* **Display Tab:** Set Framerate to `30 FPS`
* **Input Tab:** Disable the following settings:
* * `Assign multiple functions to the same input`
* * `Enhance mouse input`
* * `Enable mouse weapon swap`

**Comment by Blunderless** [comment]
> @Blunderless, we are not experiencing the same issues you are so we need your help to fix the issues. As @Polymega mentioned, attaching your log file will help us troubleshoot the issue. Also, if the issue is caused by a specific feature then knowing which feature caused the issue will help us isolate the issue and fix the issue.
> 
> Can you try disabling some of the newer features to see if the issue is in one of these features? If you figure out which feature has the issue please let us know so we can fix it.
> 
> Here are some suggestions for features to try turning off:
> 
>     * **Display Tab:** Set Framerate to `30 FPS`
> 
>     * **Input Tab:** Disable the following settings:
> 
>     * * `Assign multiple functions to the same input`
> 
>     * * `Enhance mouse input`
> 
>     * * `Enable mouse weapon swap`

aside from me having all those things active, i don't know how to send the log or which area of the log i should send as it's way to big to capture with one screenshot

**Comment by Blunderless** [comment]
> game in between

@Polymega i really don't know how to do that...

**Comment by Polymega** [comment]
There is a `d3d8.log` file in the game's main folder. Simply drag and drop that file into a reply here. Doing that will attach the log file to your post for us to look at.

-----------------------

Close out the game then launch `SH2EEconfig.exe`. This is the Config Tool. In this tool you'll see multiple tabs at the top which will take you to different "pages" within the tool. 

I'd recommend you disable everything you can within just a single page then click the "Save & Launch Game" button at the bottom right. Test the game. Does it still crash when pressing Esc? If so, re-open the Config Tool, go to a new page within this tool, and disable everything you can in that page. "Save & Launch Game" and test again. 

Do this until hopefully the issue no longer happens. If it no longer happens, you should then know what page within the Config Tool the issue is being caused at. From here, start enabling features within this page one-by-one, doing the "Save & Launch Game" in between each feature you enable, until you can pin-point what specific feature is causing the issue.

**Comment by elishacloud** [comment]
Yes, there is definitely an issue here.  The last three lines are this:

```log
13048 16:10:21.973 GetDiskSpace Available disk space smaller than 2TBs: 299406272
13048 16:10:50.190 CheckMemoryAddress Error: memory address not found!
13048 16:10:50.191 SearchAndGetAddresses searching for memory address! Found = 00476515
```

I need to add some more logging to see why this is happening.  Some recent change has the wrong address.

**Comment by elishacloud** [comment]
@Blunderless, can you try with this update and give us your log file after it crashes?  This update has extra logging so we can see where it is crashing.  Just unzip this into the Silent Hill 2 folder.

Here is the update:  [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/10912934/d3d8.zip)

**Comment by Blunderless** [comment]
> can you try with this update and give us your log file after it crashes? This update has extra logging so we can see where it is crashing. Just unzip this into the Silent Hill 2 folder.


[d3d8.log](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/10914748/d3d8.log)

thanks

**Comment by elishacloud** [comment]
@Polymega, I can reproduce this issue.  It appears that he is using the Director's Cut edition of the game.  If you use that edition of the game it crashes when you go to open the inventory.  Just FYI.

@Blunderless, can you update the game's binary file (sh2pc.exe) with the one we have posted here:  https://files.townofsilenthill.com/SH2EE/packages/SH2EE_Enhanced_EXE_NA_v1.0.zip

Just unzip this file into the Silent Hill 2 game folder.  Note: you may want to backup/rename the old file before copying this new one.

**Comment by elishacloud** [comment]
From the log file issue is in the `GetInventoryStatusPointer()` function, which is located [here](https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/41dee6f18c21ca2ff904f892281a373aa7bf6047/Patches/Common.cpp#L685-L704).

```log
11676 17:23:42.570 CheckMemoryAddress -> GetInventoryStatusPointer Error: memory address not found!
11676 17:23:42.571 SearchAndGetAddresses -> GetInventoryStatusPointer searching for memory address! Found = 00476515
```

**Comment by Polymega** [comment]
If I'm reading that code right, it's using `00476061` for 1.0, `00476301` for 1.1, and `0047605D` for DC? Here are those instructions for those addresses:

1.0
![image](https://user-images.githubusercontent.com/27395260/223589891-d0aa1114-3af3-46f2-bc34-66d855647319.png)

1.1
![image](https://user-images.githubusercontent.com/27395260/223589908-8551f260-166e-4ca5-a825-758cbe69f05f.png)

DC
![image](https://user-images.githubusercontent.com/27395260/223589926-071f9006-d788-4f3f-8ca9-8900a6907327.png)

They're all different, with DC being the most different? I might be misunderstanding something with this.

**Comment by Polymega** [comment]
For DC, it might should be `00476511`. @AeroWidescreen, may I ask you to look into this?

**Comment by elishacloud** [comment]
> For DC, it might should be `00476515` `-0x04` = `00476511`.

Yes, that looks right.   But it is still crashing with the DC version, so there is another issue besides this one.

**Comment by Polymega** [comment]
> Yes, that looks right. But it is still crashing with the DC version, so there is another issue besides this one.

Well that's a bummer. Does the new log report tell us any new info after switching out that address?

Do we know when/why `GetInventoryStatusPointer()` was created? Perhaps if we know what feature(s) use it, that could help lead us in the right direction?

**Comment by elishacloud** [comment]
It seems to be crashing somewhere inside the DirectInput8 wrapper.  It has nothing to do with the inventory.

**Comment by elishacloud** [comment]
The issue is caused by [this line of code](https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/41dee6f18c21ca2ff904f892281a373aa7bf6047/Wrappers/dinput8/IDirectInputDevice8A.cpp#L125).  What's worse is that there is no way to disable that line of code, so even if you disable all options in SH2EEConfig it will still crash.

**Comment by elishacloud** [comment]
The crash is caused by something inside the `GetPauseMenuQuitIndex()` function somewhere in [this part of the code](https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/41dee6f18c21ca2ff904f892281a373aa7bf6047/Patches/InputTweaks.cpp#L920-L946).

**Comment by elishacloud** [comment]
Ok, I put a [fix](https://github.com/elishacloud/Silent-Hill-2-Enhancements/commit/0f1fc5873bae67db749fd841b9c4073971506d5a) in for this.  Both the Director's Cut and v1.1 was completely broken and unplayable.

Here is the fixed build:  [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/10924166/d3d8.zip)

**Comment by Polymega** [comment]
Thanks, Elisha! I'm not sure everything I should be testing/looking out for, but I only found one issue so far. Left/right mouse movement isn't working while on the pause screen for the DC version:

### 1.0 & 1.1
https://user-images.githubusercontent.com/27395260/223823855-4f42f891-3e95-407c-921a-04ebdf77bad0.mp4

### DC
https://user-images.githubusercontent.com/27395260/223823887-1cc13390-9156-4f1b-9333-b705b8a4aa72.mp4

**Comment by elishacloud** [comment]
> Left/right mouse movement isn't working while on the pause screen for the DC version

This is another issue, unrelated to my fix.  Can we get @mercury501 to look into this issue?

**Comment by Blunderless** [comment]
@elishacloud  @Polymega okay i need to close this and thank y'all so effing much for actually giving shits

y'all are the best, thanks for not leaving me in the dark

**Comment by mercury501** [comment]
I'll look into it, it could be dc addreses again

**Comment by elishacloud** [comment]
Thanks @mercury501!  I think you are right about the DC address.  I figured it would be easier for you to check it out since you know that part of the code better.

**Comment by mercury501** [comment]
Hi there!

Was just a small offset change, here is the build:
[d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/10945113/d3d8.zip)
